src/main/java/org/edgexfoundry/messaging/MQTTSender.java

```diff
@@ -97,6 +101,16 @@ private void connectClient() {
 			connOpts.setPassword(password.toCharArray());
 			connOpts.setCleanSession(true);
 			connOpts.setKeepAliveInterval(keepAlive);
+			if (secureConnection) {
+				try {
+					EdgeXSecurityManager secMgr = new EdgeXSecurityManager();
+					SSLContext ctxt = SSLContext.getInstance("TLS");
+					ctxt.init(secMgr.getKeyManagers(), secMgr.getTrustManagers(), new java.security.SecureRandom());
+					connOpts.setSocketFactory(ctxt.getSocketFactory());
+				} catch (Exception e) {
+					logger.error("Failed to initialize secure connection to broker.  Will attempt unsecure connection.", e);
+				}
```

Comment:
Let's raise an MQTTException here after the error log to fail cleanly on MQTTS initialization failure.